### DISCUSSION TITLE
Project maintainership

### DISCUSSION THREAD

**Discussion by lovelydinosaur** [mkdocs/mkdocs#2442] [General]
So, @waylan has said he'll be [moving on from the project after the next release](https://github.com/oprypin/mkdocs/discussions/2#discussioncomment-745295). It's not always a very rewarding position to be in and we've been fortunate to have had his stewardship for such a long time.

It's not really clear what's going to be best for the project next. From my *personal* perspective, a project at this stage generally needs someone who's willing to turn down feature requests and anything that brings additional complexity or release-risk, and focus on just keeping things well maintained. (But other folks might have differing opinions there.)

Having said that, I could see myself being more invested if there was enough drive to push a significant new version of MkDocs, which I have explored a little in the past. I'd be curious to gather folks' thoughts on what the most significant gaps in MkDocs are at the moment?

**Comment by squidfunk** [comment]
Thanks for starting this, Tom. Personally, I think radical grooming of the issues/backlog would be a great idea to start. There're a lot of issues with "needs design decision" or "needs confirmation" labels that could easily be considered out-of-scope for this project. After the grooming, it might be more clear what this project needs.

While I'm not deeply familiar with the stock MkDocs themes, I'm happy to assist in evaluating frontend-related issues or feature requests as part of a joint session (via Zoom, or similar).

**Reply by ultrabug** [reply]
> I think radical grooming of the issues/backlog would be a great idea to start

I agree, with 1.2 out and 100+ open issues and 15+ PRs this sounds a good place to start.

What I like the most with this approach is that it sounds a perfect place to start shaping what we sould focus on next :+1: 

> I'm happy to assist in evaluating frontend-related issues or feature requests as part of a joint session (via Zoom, or similar)

Maybe the interested folks could indeed meet on a joint "issue/PR" kill session? That would make it both more human and efficient.

**Reply by lovelydinosaur** [reply]
> I think radical grooming of the issues/backlog would be a great idea to start. There're a lot of issues with "needs design decision" or "needs confirmation" labels that could easily be considered out-of-scope for this project. After the grooming, it might be more clear what this project needs.

That sounds like a great idea.

Nobody needs my permission to be able to get started on this. If anyone's up for having a comb through the issues and reviewing or suggesting closes where appropriate, then I'd be happy to add maintainer rights to them so that they can take responsibility for managing ticket statuses themselves.

Moving issues into discussions is also helpful since they then don't have a hard open/close state that strictly requires maintainer input.

**Reply by squidfunk** [reply]
To save time, IMHO, it makes sense to do this in one session. Unfortunately, I don't have the time to comment and discuss every open issue whether it needs closing in written form, but pairing with a maintainer who has the rights as part of a Zoom (?) call is something I could offer.

And yes, converting issues to discussions is also something that could be considered in this instance. Just drop me a message on Twitter if you or somebody else wants to do it and to find a time slot.

**Reply by ultrabug** [reply]
@tomchristie @squidfunk I'll be happy to participate in a joint 1-2H call to help in addressing those issues/PRs :hand: 

I agree with Martin that the participation of a maintainer is needed for this to be efficient so we can have an immediate impact on the project and not waste our time in asynchronous discussions.

**Reply by lovelydinosaur** [reply]
Okay, I'm up for giving that a go.
I'm fairly free next week, roughly within European working hours.
Martin, I've drop you a private message, to figure out a first pass onto possible time slots.
Thanks @ultrabug.

**Reply by squidfunk** [reply]
@ultrabug @tomchristie I've sent you both a link to a Doodle for coordination via Twitter.

**Reply by ultrabug** [reply]
@tomchristie we've been trying to reach out to you for a while with Martin. I can't seem to be allowed to message you on Twitter so I'd like to give it a shot here. Feel free to reach out (same nick on twitter) or tell us about your plan to move on, I'll do my best to make do.

**Comment by oprypin** [comment]
I had already offered to take over maintenance of MkDocs on 7 Apr to @waylan. I will offer it again here.

---

* I have the experience to understand and fairly evaluate requests of both users and plugin developers -- being [active on both fronts](https://github.com/search?q=oprypin+mkdocs&type=code).
* I am well familiar with most of the code (though I'll admit that I'm completely unfamiliar with the default themes).
* I have the time to drive changes that benefit convenience to contributors.
* I can offer high-quality contributions directly.
* Generally, I am an experienced software engineer, often with a focus on backwards compatibility.

Despite how it may appear, my style of maintenance is quite similar to the current one, but I put more effort trying to get closer to the objective truth.

---

If I am accepted, the basic steps would be:

* I get enough access to manage the issues and pull requests.
* I do not push changes directly.
* I do not merge pull requests before they had enough time for comments (as such, I would encourage people involved here to subscribe to pull requests).
* I respect argumented opposing opinions.

In case there are concerns of me going rogue:
_(and the reason I even have to write this is that there was not a healthy environment for a contributor to prove themselves through normal means)_

* Please look at my profile and [mentions](https://github.com/search?q=%22oprypin%22&type=Code). That is really not in my interest.
* I'm pretty sure the rights management is granular enough that @tomchristie would be able to lock me out whereas I would not be able to lock them out.

For other approaches, I am also open to discussion.

---

> From my personal perspective, a project at this stage generally needs someone who's willing to turn down feature requests and anything that brings additional complexity or release-risk, and focus on just keeping things well maintained.

This needs a *huge* qualifier. Arbitrarily denying some of users' reports is not what people need from a maintainer. As such, I intend to make things better in terms of benefit analysis and objectivity. Despite how it may appear, that definitely does not imply that I'll just approve things left and right.

**Reply by Andre601** [reply]
> For other approaches, I am also open to discussion.

The only other approach I could see here is having a fork, which while giving you the necessary rights to handle and push changes is suboptimal as it couldn't be pushed under the mkdocs PyPI namespace (or what you call it) but instead a different way. This would obviously be a downside and - unless maintainers of the original would link and recommend it - getting people to find it would be a tedious task to do.

I for my part can't say if you are the best choice for this project, mainly because I lack the knowledge in this coding language to begin with, but I feel like it would be a good idea to make you a basic member, so that you could contribute more directly, while Waylan could every now and then still check and see stuff...

On another note would I offer myself to update and maintain documentation, at least on the markdown-side of things, as this is a passion of myself. See [here](https://github.com/PlaceholderAPI/PlaceholderAPI/wiki), [here](https://docs.purrbot.site) or [here](https://github.com/Andre601/OneVersionRemake/wiki/Config) for some examples of what I do.
Only (personal) issue I have here is that enforced styling guide such as limited character length per line, which is just not good to have here and would only conflict with other's view on styling and such. At the end is Markdown like HTML: It doesn't 100% matter how long a line is because once it's rendered, it looks different...

Anyway, my offer stays, but I would then look out to perhaps redo the entirety of the docs as I feel like they could receive some changes here and there...

**Reply by oprypin** [reply]
Well I wasn't implying a fork, I had already dismissed that before as non-viable for technical reasons.

---

And indeed splitting lines in Markdown brings only disadvantages. [I don't do that](https://github.com/oprypin/mkdocs-literate-nav/blame/master/README.md#L395). But that's beside the point.

**Reply by lovelydinosaur** [reply]
@oprypin, @Andre601 I don't think we need any absolute *this/that* person is **the maintainer**, it's more about who's actually committed to getting stuck into issues, which none of us needs anyone else's permission in order to start helping out on. We can add folks onto the maintainers team whenever there's a clear block on not having sufficient folks able to okay merges etc.

It *might well* also be that it's worth bringing `mkdocs` under the `encode` organisation now, with a fairly open policy on accepting folks into the `encode/maintainers` team as needed.

**Reply by Andre601** [reply]
> @oprypin, @Andre601 I don't think we need any absolute _this/that_ person is **the maintainer**, it's more about who's actually committed to getting stuck into issues, which none of us needs anyone else's permission in order to start helping out on. We can add folks onto the maintainers team whenever there's a clear block on not having sufficient folks able to okay merges etc.
> 
> It _might well_ also be that it's worth bringing `mkdocs` under the `encode` organisation now, with a fairly open policy on accepting folks into the `encode/maintainers` team as needed.

I mean. I wouldn't say that I ever suggested it needs ONE specific person for this job. MkDocs is open source and can - like so many other open source projects - be handled by community members.

**Reply by oprypin** [reply]
Again... Would you please let me work on MkDocs? There's actually nothing happening and people are just waiting for the fixes.

**Reply by lovelydinosaur** [reply]
Sure, go right ahead. Once you've come to an issue where you need maintainer permissions in order to progress it, then let me know.

**Reply by oprypin** [reply]
@tomchristie The clear example at the moment is that I'd want to merge https://github.com/mkdocs/mkdocs/pull/2464. Also clear out some recent questions submitted as issues (or actual tiny issues).
But that's not my main point. I want to do more, just don't have the motivation in this state of uncertainty.

**Reply by lovelydinosaur** [reply]
Done.

**Reply by oprypin** [reply]
@tomchristie  Uh sorry, what's done? 😅
So far I haven't noticed anything.

**Reply by oprypin** [reply]
Oh, I finally found that I had an organization invite, only if I go to the exact page https://github.com/mkdocs -- which is quite obscure of GitHub.

Thanks for the invite, I accepted it now. As far as I see, that alone has not granted me any permissions for the MkDocs repository. So @tomchristie , when you get the opportunity, could you give me "Write" access (or whatever is appropriate) at https://github.com/mkdocs/mkdocs/settings/access
-Thanks again.

**Reply by oprypin** [reply]
@tomchristie Sorry for another ping, but I would need "Write" access in order to do anything. https://github.com/mkdocs/mkdocs/settings/access

**Reply by lovelydinosaur** [reply]
Oh right, I can see that yes.

I'm kinda thinking that a more sensible approach here might be moving `mkdocs` into the encode organisation, and having the `maintainers` team there get permissions onto it (with you added to that.)

There's a pretty big pool of folks already with access there, and I can then help streamline a process in one place.

How do we (all) feel about that idea?

**Reply by oprypin** [reply]
@tomchristie You say it streamlines things but I don't see how. Literally it's the same number of clicks to grant the permission once. And with separate orgs you don't need to worry about entrusting *everything* to *everyone*. I have not seen any intersection between active people in mkdocs/ and encode/ and I don't expect this to suddenly cause such people to appear. Isn't it more straightforward when permissions to unrelated things *aren't* conflated? To reiterate: my base argument is that all of the permission arrangements that can be done in merged orgs can also be achieved in separate orgs (maybe with more clicks) but the opposite is not true.

Anyway,.. I want to keep the MkDocs organization (again, why not just *not* initiate a migration?) and have no interest in anything from "encode". Right now it seems much simpler to just grant those permissions than plan for a reorganization (which I also just argued is more complicated even in the long term). That also doesn't preclude the migration if it's later decided upon.

**Reply by oprypin** [reply]
I would like to say: this topic has been on my mind the entire time. So I'd appreciate that when you do make a reply, it doesn't leave me hanging for yet another week.

**Reply by squidfunk** [reply]
@tomchristie PRs are piling up and there are some issues with 1.2.1 which we need to get fixed. I'd be all for giving @oprypin write access so he can go ahead and push MkDocs forward. When further contributors are interested in joining, we could evaluate moving MkDocs under the Encode organization, but IMHO for now, an active maintainer like @oprypin would be the biggest gain for this project.

**Reply by ultrabug** [reply]
@tomchristie if you'd rather have two contributors added to the project and nobody objects I can propose my help so that @oprypin is not alone to face the task. I will not merge anything that's not been discussed and approved ofc... my track record speaks for itself I hope.

**Reply by lovelydinosaur** [reply]
I'm good with both you joining. Having weekend time at the moment, intend to sort it on Monday. 👍

**Reply by lovelydinosaur** [reply]
@oprypin @squidfunk @ultrabug I've sent you all team invites.

**Reply by oprypin** [reply]
Thanks!

OK for now I closed some issues that are trivial to close and merged trivial PRs.

Going further, I'm open to suggestions whether the others want to be involved in reviews or not. Generally: should I expect voluntary fly-by reviews and/or should I explicitly ask for reviews? (the alternative being that I just do whatever seems not controversial)

The other thing that I was going to do is reply on pull requests, saying that I intend to merge them in N days. So, should I do that? Will any contributor notice/care about that? 
@ultrabug  @squidfunk 

Anyone interested in having a chat room? I was thinking that even a public Gitter room would easily be able to house development discussions as well as eventual general participation.

**Reply by squidfunk** [reply]
> Going further, I'm open to suggestions whether the others want to be involved in reviews or not. Generally: should I expect voluntary fly-by reviews and/or should I explicitly ask for reviews? (the alternative being that I just do whatever seems not controversial)

I'm happy I'm on the team now, but I can't make time for regularly creating or reviewing PRs, because I'm pretty involved with Material for MkDocs and this project eats up all of my available spare time. However, as it was already done in the past, if you feel my input would be helpful on some specific topic, feel free to mention me. After all, I'm probably one of the most active MkDocs theme developers.

> The other thing that I was going to do is reply on pull requests, saying that I intend to merge them in N days. So, should I do that? Will any contributor notice/care about that?

In order to give a time window for review? I think that's cool, but maybe @ultrabug who did some work on the core has an opinion on that one. In general, I won't be popping in regularly, only when I'm mentioned.

> Anyone interested in having a chat room? I was thinking that even a public Gitter room would easily be able to house development discussions as well as eventual general participation.

A public Gitter sounds perfect. My experience with hosting one for Material for MkDocs is pretty good, so +1.

---

I skimmed through the backlog and (IMHO) still see a lot of resolved or issues or feature requests that are out-of-scope. I'm again offering to do a 1-2h video call to go through the backlog and give input on issues from my role as a theme developer.

**Reply by ultrabug** [reply]
> Going further, I'm open to suggestions whether the others want to be involved in reviews or not. Generally: should I expect voluntary fly-by reviews and/or should I explicitly ask for reviews? (the alternative being that I just do whatever seems not controversial)

If you feel this is something with a strong impact on the project, yes ask for a review. "Non controversial" is subjective anyway so as long as we can keep a positive discussion, even after the merge, is what's the most important to me.

> The other thing that I was going to do is reply on pull requests, saying that I intend to merge them in N days. So, should I do that? Will any contributor notice/care about that?

I like the idea yes. I will get those notifications and will do my best to reply there.

> Anyone interested in having a chat room? I was thinking that even a public Gitter room would easily be able to house development discussions as well as eventual general participation.

Very much interest yes. I'm on IRC all the time (OFTC and Libera.chat), I never used Gitter but if you guys did I'll join whatever you settle for. But please, allow us to have a direct communication channel somewhere!

**Reply by oprypin** [reply]
Alright, so https://gitter.im/mkdocs/community is the Gitter / Matrix channel. For now it's announced only here, but everyone is welcome!

**Reply by squidfunk** [reply]
> @oprypin @squidfunk @ultrabug I've sent you all team invites.

Today I received an email that I was removed from the MkDocs organization. This happened after my rights were degraded from maintainer to member a while ago. There was no point of contact by a maintainer prior to this action. Could somebody offer an explanation why this happened? I was happy to contribute in the past and I'm not sure what my wrongdoing is.

**Reply by oprypin** [reply]
Hi. I removed you from the GitHub organization. I am not comfortable with having it appear like you represent MkDocs (specifically on GitHub, I think the "Contributor" label is more accurate than "Member").

**Reply by pawamoy** [reply]
@oprypin I am not comfortable with the fact that you decide to remove people from the organization without notice and without discussing it with other members. Especially when it is based on "appearance", which is highly subjective (you are assuming that the community perceives @squidfunk's messages or his "Member" label as suggesting that he represents MkDocs). Though to be honest, I do think Martin represents MkDocs, just like all members and contributors do, so I think he deserves to keep his place as a member of the organization.

**Reply by Andre601** [reply]
Well, I can understand oprypin's action here. When I see "Member", I do assume they are part of the org. And usually, being part of an Org means they are also part of a form of team of said org, so even if they don't want to, they may be seen as representing the org itself through that.

Tho, frankly, everyone has their own views on this, so we shouldn't assume one specific case being the norm here.

On another note, I wonder, @oprypin have there been specific cases/situations that make you remove squidfunk?
Given your reply that you're not comfortable with him appearing like representing MkDocs on GitHub by being a Member, I assume there must've been something that you consider bad for MkDocs?
It's a bit vague, which is why I'm asking you about the exact reasons here.

**Reply by Andre601** [reply]
I just saw [your post on fosstodon](https://fosstodon.org/@squidfunk/112102851027662960) @squidfunk and I must say, that is a really low blow man:

![grafik](https://github.com/mkdocs/mkdocs/assets/11576465/f65587fe-ce1a-4be2-af37-03a724ef5c03)

I can only assume this is in reference to MkDocs, as I'm not aware of you being removed from any other Org.
So you posting a comment like this, making it seem like you are the victim because you apparently "make a successful project". This feels really bad man.
I wouldn't be surprised if it was comments like these that are the reason for your removal. Because it certainly wouldn't be the first time where you blame others for things, like constantly claiming non-sponsors are demanding in every aspect and not grateful for the work you did.

Also, this post was made 12 hours ago (Based on me posting this comment now), compared to his comment here being from 9 hours ago... So he made the post before even commenting here first.
And you really have the audacity to ask if we can act like adults in your post, while doing this?

**Reply by pawamoy** [reply]
@Andre601 you don't seem to be aware of the whole context. I invite you to wait for more details from @oprypin before taking a side. You can also (re)read public interactions between @oprypin and @squidfunk in this very repository. It is not black and white, of course, but I can understand that Martin is tired by the situation (I am too) and needs to blow off some steam on Mastodon.

**Reply by Andre601** [reply]
It is understandable, but still not justifying a behaviour like this on other platforms. Like with his post, he clearly pushes the idea that he was kicked for having a successful project, which certainly isn't the reason here at all...

Like, I thought open source is meant to be inviting and supportive, not shift the blame to others and make them look that bad, only to boost yourself.

**Reply by squidfunk** [reply]
1. I kept my membership in the MkDocs organization private at all times – I've never disclosed it publicly. I never _represented_ MkDocs publicly. Maintainers could see that I was a member, but it was not visible to anyone else outside of the organization. The documentation of Material for MkDocs clearly attributes MkDocs as the underlying framework.
2. Yes, my Twitter/Fosstodon post was related to this action and the fact that I'm a little annoyed about this. I'm sorry to say, but I also get annoyed at times, I'm a human being after all. I did, however, not disclose what this was about, only my learning that some people will give you a hard time when you (have to) maintain something successful. Additionally, it is the distillation of countless attempts to talk to @oprypin. I have given up on doing that privately, as we were running around in circles, as you can read in [my comment here](https://github.com/mkdocs/mkdocs/issues/3579#issuecomment-1970691690), after he made me to. I've tried it publicly but I was ghosted in all attempts. I'm honestly at the end of ideas. If you have an idea how we can resolve this, go ahead. I'm all yours.
3. I appreciate all users that use my work. However, sometimes I feel that it's fair to point out that the work we Open Source maintainers do, tirelessly fixing bugs, helping users and answering questions, is not appreciated enough. Many Open Source projects fall apart because of this, maintainers burn out. I feel that I have the obligation to share this with the wider community and how hard maintaining a project of this size is. You can find countless blog posts by other maintainers that burned out because of this. I fight very, _very_ hard not to. This is my way of coping with it.

**Reply by squidfunk** [reply]
> [...] Like with his post, he clearly pushes the idea that he was kicked for having a successful project, which certainly isn't the reason here at all... [...] Like, I thought open source is meant to be inviting and supportive, not shift the blame to others and make them look that bad, only to boost yourself.

The post is the result on many, many private and public conversations. What @oprypin is referring to in his comment is something else entirely I do not understand, because as I said in 1., I have never said that I'm representing MkDocs. I'd be interested in seeing what exactly you or @oprypin thinks looks like I'm representing MkDocs.

**Reply by oprypin** [reply]
Hi @squidfunk .

1. You being in the repository had no functional reason, so I made this more accurately reflected.

   > Maintainers could see that I was a member, but it was not visible to anyone else outside of the organization.

   Oh I see now. I checked this now. Yes so the label on the comments was actually not even visible to others. So then removing you wasn't necessary for this!

3. I don't know what you're trying to say in that paragraph ("look what he made me do"?), so just to clarify:

   Half a year ago, you blocked me in chat, removed me from your repository (to which I *was* contributing and was relying on to check against the ecosystem), and then you're saying that you made all attempts to "resolve this".
  Yes, unsurprisingly, I'm not interested in pouring garbage publicly.
  No, there wasn't any reason to block me in chat.

**Reply by Andre601** [reply]
> This is in reply to [this post](https://github.com/mkdocs/mkdocs/discussions/2442#discussioncomment-8812492) and not any of the follow-ups.

I can understand it's frustrating. I do open source work too and often have people of the "frustrating" kind, aka. asking questions answered within documentation, yet I still don't go on social media and blame an entire group (in your case non-sponsors) for something maybe a third does here.

Do I get mad? Yes. Does it tire me? Also yes. But when this happen do I simply take a break, get myself a coffee and try to relax for a bit before getting back and trying to resolve it.
That being said, at times am I also reacting less than apropriate to some support requests. But that is usually after I spent a lot of time and energy on helping the person already and they still don't try to understand it.

Also, I do this while I don't earn a single dime AND have financial struggles... Just to give a perspective here.

Like, you [earn $15,450 a month](https://squidfunk.github.io/mkdocs-material/insiders/#funding) which unless you share some clear cut info on how much each of your [mentioned payments](https://squidfunk.github.io/mkdocs-material/insiders/faq/sponsoring/#sponsorship-support) uses up, I'll assume you still gain around 3k a month for yourself (After taxes and all would be removed), which would allow you to take breaks much more often when having tiring support to deal with...
(This is btw why I made a discussion, [asking for more clarity](https://github.com/squidfunk/mkdocs-material/discussions/6696) to avoid guesswork and amiguity here...)

**Reply by pawamoy** [reply]
@Andre601 you are free to handle the frustrations of open-source the way you want. Yet that doesn't mean everybody else has to handle them the way you do.

You seem to imply that because someone is getting paid or makes money from their work in open-source, they somehow must accept unpleasant situations and refrain from sharing their experience publicly.

Also, IMO the sponsorware stragey is already one of the most *transparent* model there is, yet you still ask for more information, which by the way can be private and/or sensitive. That sounds... entitled. And what does this have to do with the situation here anyway?

**Reply by Andre601** [reply]
> You seem to imply that because someone is getting paid or makes money from their work in open-source, they somehow must accept unpleasant situations and refrain from sharing their experience publicly.

The main issue here is, that Squidfunk, through his messages, is representing his Project, which may not be in a positive light.
How much would you like a project, who's core maintainer and owner is complaining about people on social media?

> Also, IMO the sponsorware stragey is already one of the most transparent model there is, yet you still ask for more information, which by the way can be private and/or sensitive. That sounds... entitled. And what does this have to do with the situation here anyway?

Where in the world is sharing percentages a private or sensitive info? In my discussion did I even state that I don't ask for outright numbers, but ther overall percentages here.

Also, my point here was, that he is certainly earning enough to take a small break at times from giving support to not burn out, without having to fear financial issues...
Yes, it's not really fitting for this conversation, but frankly, this entire discussion here is not even fitting for this original post itself, so it probs is best to move away from here to a different place to resolve it between squidfunk and oprypin.

**Reply by squidfunk** [reply]
> Yes, unsurprisingly, I'm not interested in pouring garbage publicly. [...] No, there wasn't any reason to block me in chat.

We talked about this many, many times, privately. After this was going nowhere, as you kept your position no matter what I said, I decided to limit conversation to the public space. I didn't simply make this decision, _other people recommended it to me_. You also repeatedly claim other things publicly thatnyou said to me in private. It _is_ frustrating. Blocking you on Gitter was my very, very last resort because you kept giving me a hard time. Despite blocking you, I kept engaging in constructive and friendly public conversation in several issues, some of which I submitted PRs for.

As I wrote in the comment I linked, I still believe in the quality of your work.

> Also, IMO the sponsorware stragey is already one of the most transparent model there is, yet you still ask for more information, which by the way can be private and/or sensitive. That sounds... entitled. And what does this have to do with the situation here anyway?

Nothing more to add.

**Reply by squidfunk** [reply]
@Andre601 you are making a lot of assumptions about my private life and situation. If I earn money from my Open Source work, do I have the obligation to be entirely transparent? I am not going to disclose private matters.

**Reply by pawamoy** [reply]
@Andre601 Martin complains about entitled users and people giving him a hard time because he earns money thanks to his open source work. I don't see how that's an issue. If anything, I relate to that.

Each person decides what is private information. They don't have to share any of it when someone asks them, unless they're the government. Numbers are *instantly* derived from percentages. I don't understand why you make a distinction between the two.

I think it's that kind of comments that gives people a hard time when they have a successful project, or one that allows them to earn money: somehow they have to comply with demands from random persons, and they also can't complain because they earn money so surely they can take breaks. IMO these comments actually explain Martin's posts on social media.

**Reply by lovelydinosaur** [reply]
I'm stepping in here as the original author, in a mediator role.
For the moment I've ensured that there's no changes in membership status; since I'm uncomfortable with the direction that took.
I'd suggest we pause conversation on this thread so that there's a cooling off period before we try to figure out what's not working smoothly here.

**Reply by oprypin** [reply]
@tomchristie So your way of de-escalating the situation is to simply remove rights from me, without any follow-up? That is completely unnecessary, I couldn't even imagine something like that.

**Reply by squidfunk** [reply]
Thanks @tomchristie for reinstating my member status ✌️ Let's all cool down and resolve this in a constructive fashion ☺️

**Reply by lovelydinosaur** [reply]
Okay, I can see some communication issues all round here. As it stands I don't think there's any issues currently blocked by existing team members requiring "ownership" permissions, so I'm going to hold onto the ownership keys at this point in time. If there are specific organisational issues that require resolving please just ask.

I've created https://github.com/mkdocs/mkdocs-team-space as a space for us to have discussions around project maintainership etc. All existing team members currently have "maintain" permissions onto that repo, I think that ought to allow y'all to invite other plugin authors and invested parties into the space.

**Reply by oprypin** [reply]
My response:

* https://github.com/mkdocs/mkdocs/discussions/3621

**Reply by Andre601** [reply]
By the way, I would like to point out one small thing that seems to have changed since the last conversations here.

Namely, it was pointed out that @squidfunk's relationship with the org (That he is part of the team) was not publicly shown... This seems to have changed, because he and other members of the org (such as oprypin, ultrabug and tomchristie) now display a "Maintainer" tag next to their names in discussions (And I assume issues and PRs, but I haven't checked it yet).

I assume this is a result from recent role changes in the org? If so, then I can see Oprypin's original worries reignited as people may see the Maintainer tag and automatically associate the person's actions with the project, even if they may not be related to the project itself...

Just wanted to point that out.

**Reply by Andre601** [reply]
Also, I have to say this, but [this line kinda aged like milk](https://github.com/mkdocs/mkdocs/discussions/2442#discussioncomment-857569) as you @tomchristie seem to now take this "this/that" person role you talked about wasn't necessary in the past... Oh how views change over time.

> [@]oprypin, [@]Andre601 I don't think we need any absolute this/that person is the maintainer, it's more about who's actually committed to getting stuck into issues, which none of us needs anyone else's permission in order to start helping out on.

**Reply by squidfunk** [reply]
@Andre601 I asked about this on [GitHub's private maintainers community](https://github.com/community/maintainers/discussions/385), as it is in fact showing the maintainer badge although I had my settings set to private (as I do now). No reaction from GitHub yet. It also sounds like an error to me.

Additionally, I second Tom here. Everybody that is a member can start working on moving the project forward – we can merge PRs, make releases and triage issues. There's no need to be an owner right now, and it shows Tom's trust in all of us.

**Comment by ultrabug** [comment]
I'm away on vacation sorry, will reply on July 12, thanks mate

**Comment by astrojuanlu** [comment]
I see that @oprypin [published 1.2.2](https://github.com/mkdocs/mkdocs/releases/tag/1.2.2). Congrats! Does that mean that the project maintainership is sorted out? (Didn't see any further updates here)

**Reply by oprypin** [reply]
Basically yes.

https://github.com/mkdocs/mkdocs/discussions/2442#discussioncomment-964629

**Reply by astrojuanlu** [reply]
Oh, those were hidden by default. Thanks, and congrats!

**Comment by oprypin** [comment]
Hi!
@tomchristie @waylan 

Could I please get "Owner" rights for the [MkDocs organization](https://github.com/orgs/mkdocs/people)?

I would be able to
• change settings of repositories (right now the "Settings" tab is not accessible at all!)
• add others to repositories
• block commenters ([example](https://github.com/mkdocs/mkdocs/pull/2852#issuecomment-1144306688))
• configure / remove bots ([example](https://github.com/mkdocs/mkdocs/pull/2628))
• delete double-posted issues
• safely adopt new repositories (recently a new repository [was transferred](https://github.com/mkdocs/mkdocs-redirects/issues/32#issuecomment-1079737699) to the organization, but it had to first be transferred to me and then I transferred to MkDocs at which point I can't even transfer it out if I wanted)

---

**Additionally** and possibly more importantly, I probably should be added to [the PyPI project](https://pypi.org/manage/project/mkdocs/collaboration/). So far releases have been triggered by pushing to GitHub and having it use the token (that I can't see or configure), but that's fragile if anything ever goes wrong.

**Reply by lovelydinosaur** [reply]
Done. You'll need to prompt me about the PyPI access some other time, can't do that right now.

**Reply by oprypin** [reply]
Thanks much!

**Comment by pawamoy** [comment]
After some discussion it's been decided to try and onboard more plugin/extension authors on the team, see our announcement here: https://github.com/mkdocs/mkdocs/discussions/3622. I just thought like sharing this here since it's relevant to the project's maintainership :slightly_smiling_face:
